cirq-core/cirq/sim/act_on_args.py

```diff
@@ -77,7 +80,7 @@ def __init__(
                 for k, v in (log_of_measurement_results or {}).items()
             }
         )
-        self._state = state
+        self._state = cast(TState, state)
```

Comment:
Can we specify `state: TState` in the arg list instead of casting? Or does that break the callers?

Reply:
I think it has to be a cast because it's an optional parameter. Ideally we could make it non optional now, but it occurs after some other optional parameters in the method signature. That's part of why I deprecated positional arguments in the other PR.